src/typeconversions.jl

```diff
@@ -60,6 +60,15 @@ function datatype(str::VLen{C}) where {C<:CharType}
     Datatype(API.h5t_vlen_create(type_id))
 end
 
+# Compound types
+function datatype(A::AbstractArray{T}) where {T}
```

Comment:
Since you're just creating the compound type rather than array type, the argument should probably just be the type itself. You may then want to manually dispatch based on whether `isstruct` is `true`.

Reply:
Make sense... However, I hit the wall somewhere because `Type Array does not have a definite size.`, which I do not understand yet ;)

Reply:
You need something from StaticArrays.jl for the size to be in the type.

Otherwise, you can just take an instance and then dispatch on `isstruct`.